### PR TITLE
openjdk/darwin: move version out of name

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/openjdk/darwin/default.nix [hunk 1]
```diff
@@ -7,10 +7,11 @@ let
   };
 
   jdk = stdenv.mkDerivation rec {
-    name = "zulu15.28.51-ca-jdk15.0.1";
+    pname = "zulu15.28.51-ca-jdk";
+    version = "15.0.1";
 
     src = fetchurl {
-      url = "https://cdn.azul.com/zulu/bin/${name}-macosx_x64.tar.gz";
+      url = "https://cdn.azul.com/zulu/bin/${pname}-${version}-macosx_x64.tar.gz";
```
Comment (sternenseemann) [inline pkgs/development/compilers/openjdk/darwin/default.nix:14]: The URL is this:

```suggestion
      url = "https://cdn.azul.com/zulu/bin/${pname}${version}-macosx_x64.tar.gz";
```

Is suspect this prompted the awkward naming choice in the first place

**Review by sternenseemann — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/openjdk/darwin/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/development/compilers/openjdk/darwin/default.nix]: Thanks.

File: pkgs/development/compilers/openjdk/darwin/default.nix [hunk 2]
```diff
@@ -7,10 +7,11 @@ let
   };
 
   jdk = stdenv.mkDerivation rec {
-    name = "zulu15.28.51-ca-jdk15.0.1";
+    pname = "zulu15.28.51-ca-jdk";
+    version = "15.0.1";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/darwin/default.nix:11]:
```suggestion
    pname = "zulu";
    version = "15.28.51-ca-jdk15.0.1";
```

maybe? but I am not sure.

Also just found the same bug.

Comment (siraben) [reply pkgs/development/compilers/openjdk/darwin/default.nix:11]: What bug? jdk seems to work now
```
nix-repl> jdk.version
"15.0.1"
```

**Comment by siraben** [comment]
250 rebuilds but it's mostly harmless™

**Comment by SuperSandro2000** [comment]
Verified that openjdk and vscode-extensions.redhat.java now builds locally.

**Pull request merged**